jerry-core/ecma/base/ecma-helpers-conversion.c

```diff
@@ -620,6 +640,22 @@ ecma_utf8_string_to_number (const lit_utf8_byte_t *str_p, /**< utf-8 string */
     return sign ? -ECMA_NUMBER_ZERO : ECMA_NUMBER_ZERO;
   }
 
+  /** If the convertable number is bigger than Number.MAX_VALUE
+    * then return with +- Infinity */
+  if (!e_in_lit_sign && (e > NUMBER_MAX_VALUE_EXPONENT ||
```

Comment:
Should this be `e_sign` or `e_in_lit_sign`? We compute `e_in_lit` and `e_in_lit_sign` first, then adjust and put them into `e` and `e_sign`. In this condition we seem to mix the two sets of values: we use `e` but wit `e_in_lit_sign`. Seems suspicious.